<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to MarZIC</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to MarZIC</h1>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>MarZIC estimates and tests the marginal mediation effects of
zero-inflated compositional mediators. It can be applied to microbiome
studies to estimate and test the marginal mediation effects of relative
abundances (RA). Let <span class="math inline">\(Y\)</span>, <span class="math inline">\(M\)</span> and <span class="math inline">\(X\)</span> denote the outcome, mediator (RA) and
the independent variable respectively. And let <span class="math inline">\(1_{(M&gt;0)}\)</span> denote the binary indicator
variable indicating whether <span class="math inline">\(M\)</span> is
positive. MarZIC decomposes the mediation effect into two components of
which one is the total mediation effect going thru the continuous scale
of the mediator, denoted by NIE<span class="math inline">\(_1\)</span>
and the other is the mediation effect going thru the binary change (from
zero to non-zero status) of the mediator, denoted by NIE<span class="math inline">\(_2\)</span>. For a microbiome data, let <span class="math inline">\(M_j\)</span> denote the RA of the <span class="math inline">\(j\)</span>th taxon/OTU/ASV. And let <span class="math inline">\(Z\)</span> denote confounder variable (if any) and
<span class="math inline">\(Z\)</span> could contain multiple variables.
The marginal outcome model for <span class="math inline">\(Y\)</span>
is: <span class="math display">\[
Y=\beta_0+\beta_1M_j+\beta_21_{M_j&gt;0}+\beta_3X+\beta_4X1_{M_j&gt;0}+\beta_5XM_j+\beta_6Z+\epsilon
\]</span> The mean of <span class="math inline">\(M_j\)</span> depends
on <span class="math inline">\(X\)</span> and <span class="math inline">\(Z\)</span> through the following equation: <span class="math display">\[
\log \Bigg(\frac{E(M_j)}{1-E(M_j)}\Bigg)=\alpha_0 + \alpha_1X+\alpha_2Z
\]</span> Let <span class="math inline">\(\Delta_j=P(M_j=0)\)</span>
which is the probability of a structural zero (ie, true zero). We use
the following logistic model for this probability:<br />
<span class="math display">\[
\log\bigg(\frac{\Delta_j}{1-\Delta_j}\bigg)=\gamma_0 +
\gamma_1X+\gamma_2Z
\]</span></p>
</div>
<div id="input-for-marzic-function" class="section level1">
<h1>Input for MarZIC() function</h1>
<p>Typically, users just need to feed the first seven inputs to the
function: <code>MicrobData</code>,<code>CovData</code>,
<code>lib_name</code>, <code>y_name</code>, <code>x_name</code>,
<code>conf_name</code> and <code>taxa_of_interest</code>.</p>
<ul>
<li><code>MicrobData</code> A dataset contains microbiome data. The
microbiome data could be relative abundance or absolute abundance.
Subjects with missing value will be removed during analysis.</li>
<li><code>CovData</code> A dataset contains outcome, library size and
covariates.</li>
<li><code>lib_name</code> Name of library size variable.</li>
<li><code>y_name</code> Name of outcome variable</li>
<li><code>x_name</code> Name of covariate of interest</li>
<li><code>conf_name</code> Name of confounders Defaule is NULL, meaning
no confounder.</li>
<li><code>x4_inter</code> Whether to include the interaction term <span class="math inline">\(\beta_4\)</span>. Default is TRUE.</li>
<li><code>x5_inter</code> Whether to include the interaction term <span class="math inline">\(\beta_5\)</span>. Default is TRUE.</li>
<li><code>taxa_of_interest</code> A character vector for taxa names
indicating taxa that should be analyzed. Default is “all”, meaning all
taxa should be included into analysis.</li>
<li><code>mediator_mix_range</code> Number of mixtures in mediator.
Default is 1, meaning no mixture.</li>
<li><code>transfer_to_RA</code> Logical variable indicating whether the
microbiome data should be transferred into relative abundance. Default
is TRUE. If FALSE, microbiome data will not be transferred (e.g., if the
input data is already RA data).</li>
<li><code>num_cores</code> Number of CPU cores to be used in
parallelization task.</li>
<li><code>adjust_method</code> P value adjustment method. Same as
p.adjust in R. Default is “fdr”.</li>
<li><code>fdr_rate</code> FDR cutoff for significance. Default is
0.05.</li>
<li><code>taxDropThresh</code> The threshold of dropping taxon due to
high zero percentage. Default is 0.9, meaning taxon will be dropped for
analysis if zero percentage is higher than 90%.</li>
<li><code>taxDropCount</code> The threshold of dropping taxon due to not
enough non-zero observation counts. Default is 4 *
(length(conf_name)+2), meaning taxon will be dropped if non-zero
observation is less than four times of the total number of covariates
(including the independent variable) plus intercept.</li>
<li><code>zero_prop_NIE2</code> The threshold of zero percentage for
calculating NIE2. Default is 0.1, meaning NIE2 will be calculated for
taxon with zero percentage greater than 10%.</li>
<li><code>zero_count_NIE2</code> The threshold of zero counts for
calculating NIE2. Default is 4 * (length(conf_name)+2), meaning NIE2
will be calculated for taxon with zero counts greater than four times of
number of covariates plus 1.</li>
<li><code>SDThresh</code> The threshold of dropping taxon due to low
coefficient of variation (CV) to avoid constant taxon. Default is 0.05,
meaning any taxon has CV less than 0.05 will be dropped.</li>
<li><code>SDx</code> The threshold of stopping analysis due to low CV of
covariate of interest. Default is 0.05, meaning when CV of covariate of
interest is less than 0.05, the analysis will be stopped.</li>
<li><code>SDy</code> The threshold of stopping analysis due to low CV of
outcome. Default is 0.05, meaning when CV of outcome is less than 0.05,
the analysis will be stopped.</li>
</ul>
</div>
<div id="output-for-marzic-function" class="section level1">
<h1>Output for MarZIC() function</h1>
<p>A <code>list</code> of <code>4</code> datasets containing the results
for <code>NIE1</code>, <code>NIE2</code>, <code>NDE</code>, and
<code>NIE</code>. Each dataset has row representing each taxon, 6
columns for <code>Estimates</code>, <code>Standard Error</code>,
<code>Lower bound for 95% Confidence Interval</code>,
<code>Upper bound for 95% Confidence Interval</code>,
<code>Adjusted p value</code>, <code>Significance indicator</code>.</p>
</div>
<div id="example" class="section level1">
<h1>Example</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="do">## A make up example with 1 taxon and 100 subjects.</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>nSub <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>nTaxa <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="do">## generate covariate of interest X</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(nSub, <span class="dv">1</span>, <span class="fl">0.5</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="do">## generate confounders</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>conf1<span class="ot">&lt;-</span><span class="fu">rnorm</span>(nSub)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>conf2<span class="ot">&lt;-</span><span class="fu">rbinom</span>(nSub,<span class="dv">1</span>,<span class="fl">0.5</span>)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="do">## generate mean of each taxon. All taxon are having the same mean for simplicity.</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="dv">5</span> <span class="sc">+</span> X <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf1 <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf2) <span class="sc">/</span> </span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="dv">5</span> <span class="sc">+</span> X <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf1 <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf2))</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>phi <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="do">## generate true RA</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>M_taxon<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">sapply</span>(mu,<span class="cf">function</span>(x) dirmult<span class="sc">::</span><span class="fu">rdirichlet</span>(<span class="at">n=</span><span class="dv">1</span>,<span class="fu">rep</span>(x<span class="sc">*</span>phi,nTaxa))))</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>P_zero <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="dv">3</span> <span class="sc">+</span> <span class="fl">0.3</span> <span class="sc">*</span> X <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf1 <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf2) <span class="sc">/</span> </span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="dv">3</span> <span class="sc">+</span> <span class="fl">0.3</span> <span class="sc">*</span> X <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf1 <span class="sc">+</span> <span class="fl">0.1</span> <span class="sc">*</span> conf2))</span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a>non_zero_ind <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">sapply</span>(P_zero,<span class="cf">function</span>(x) <span class="dv">1</span><span class="sc">-</span><span class="fu">rbinom</span>(nTaxa,<span class="dv">1</span>,<span class="fu">rep</span>(x,nTaxa))))</span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>True_RA<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">apply</span>(M_taxon<span class="sc">*</span>non_zero_ind,<span class="dv">1</span>,<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)))</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="do">## generate outcome Y based on true RA</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>Y <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">+</span> <span class="dv">100</span> <span class="sc">*</span> True_RA[,<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">5</span> <span class="sc">*</span> (True_RA[,<span class="dv">1</span>] <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">+</span> X <span class="sc">+</span> conf1 <span class="sc">+</span> conf2 <span class="sc">+</span> <span class="fu">rnorm</span>(nSub)</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="do">## library size was set to 10,000 for all subjects for simplicity.</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>libsize <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="do">## generate observed RA</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a>observed_AA <span class="ot">&lt;-</span> <span class="fu">floor</span>(M_taxon<span class="sc">*</span>libsize<span class="sc">*</span>non_zero_ind)</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>observed_RA <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(observed_AA,<span class="dv">1</span>,<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)))</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="fu">colnames</span>(observed_RA)<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">&quot;rawCount&quot;</span>,<span class="fu">seq_len</span>(nTaxa))</span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a>CovData <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">Y =</span> Y, <span class="at">X =</span> X, <span class="at">libsize =</span> libsize, <span class="at">conf1 =</span> conf1, <span class="at">conf2 =</span> conf2)</span></code></pre></div>
<p>Suppose we’re interested in the mediation effects of
<code>rawCount1</code>, <code>rawCount2</code>, and
<code>rawCount3</code>, the analysis could be done as:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">## run the analysis</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">MarZIC</span>(</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">MicrobData =</span> observed_RA,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">CovData =</span> CovData,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">lib_name =</span> <span class="st">&quot;libsize&quot;</span>,</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">y_name =</span> <span class="st">&quot;Y&quot;</span>,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">x_name =</span> <span class="st">&quot;X&quot;</span>,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">conf_name =</span> <span class="fu">c</span>(<span class="st">&quot;conf1&quot;</span>,<span class="st">&quot;conf2&quot;</span>),</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">taxa_of_interest =</span> <span class="fu">c</span>(<span class="st">&quot;rawCount1&quot;</span>,<span class="st">&quot;rawCount2&quot;</span>,<span class="st">&quot;rawCount3&quot;</span>),</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">num_cores =</span> <span class="dv">1</span>,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  <span class="at">mediator_mix_range =</span> <span class="dv">1</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>)</span></code></pre></div>
<p>The results contain 4 datasets for NIE<span class="math inline">\(_1\)</span>, NIE<span class="math inline">\(_2\)</span>, NDE, NIE, respectively. The NIE<span class="math inline">\(_1\)</span>, for example, could be extracted
by:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>NIE1 <span class="ot">&lt;-</span> res<span class="sc">$</span>NIE1_save</span></code></pre></div>
<p>The significant result could be extracted by:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">subset</span>(NIE1, significance <span class="sc">==</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt;              est      se   CI low    CI up p value unadj p value adj</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; rawCount1 7.1385 1.66603 3.873081 10.40392  1.829557e-05 5.48867e-05</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;           significance</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; rawCount1         TRUE</span></span></code></pre></div>
<p>From the result, we can find that <code>rawCount1</code> is an
significant mediator of mediating the effect of <span class="math inline">\(X\)</span> on <span class="math inline">\(Y\)</span>.</p>
</div>
<div id="reference" class="section level1">
<h1>Reference</h1>
<p>Wu et al.(2022) MarZIC: A Marginal Mediation Model for Zero-Inflated
Compositional Mediators with Applications to Microbiome Data. Genes
2022, 13, 1049.</p>
<pre><code>#&gt; R version 4.5.0 (2025-04-11 ucrt)
#&gt; Platform: x86_64-w64-mingw32/x64
#&gt; Running under: Windows 10 x64 (build 19045)
#&gt; 
#&gt; Matrix products: default
#&gt;   LAPACK version 3.12.1
#&gt; 
#&gt; locale:
#&gt; [1] LC_COLLATE=C                          
#&gt; [2] LC_CTYPE=English_United States.utf8   
#&gt; [3] LC_MONETARY=English_United States.utf8
#&gt; [4] LC_NUMERIC=C                          
#&gt; [5] LC_TIME=English_United States.utf8    
#&gt; 
#&gt; time zone: America/New_York
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] MarZIC_1.0.1
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] doParallel_1.0.16 cli_3.6.5         knitr_1.50        rlang_1.1.6      
#&gt;  [5] xfun_0.52         Formula_1.2-4     jsonlite_1.7.2    dirmult_0.1.3-5  
#&gt;  [9] zoo_1.8-11        NlcOptim_0.6      nnet_7.3-20       htmltools_0.5.2  
#&gt; [13] pracma_2.4.2      lmtest_0.9-40     stats4_4.5.0      sass_0.4.0       
#&gt; [17] modeltools_0.2-23 rmarkdown_2.29    quadprog_1.5-8    grid_4.5.0       
#&gt; [21] evaluate_0.16     jquerylib_0.1.4   MASS_7.3-65       fastmap_1.1.0    
#&gt; [25] foreach_1.5.1     yaml_2.2.1        compiler_4.5.0    codetools_0.2-20 
#&gt; [29] mathjaxr_1.4-0    sandwich_3.0-2    Rcpp_1.0.9        rstudioapi_0.13  
#&gt; [33] flexmix_2.3-18    lattice_0.22-6    digest_0.6.29     R6_2.5.1         
#&gt; [37] parallel_4.5.0    bslib_0.3.1       tools_4.5.0       iterators_1.0.13 
#&gt; [41] betareg_3.1-4</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
